vllm/model_executor/models/baichuan.py

```diff
@@ -268,6 +268,13 @@ def forward(
         attn_metadata: AttentionMetadata,
     ) -> torch.Tensor:
         hidden_states = self.embed_tokens(input_ids)
+        if hasattr(self, 'prefix_encoder'):
```

Comment:
can we set `self.prefix_encoder` as None by default and then turn this check into an `is not None`?

also, if this logic is going to be the same for every model, can we separate it out into a function and reuse?

Reply:
1. Sure that sounds good.
2. Yes we can do that, we can abstract this out into a function which takes in the embeddings and positions and spits out the new hidden states